src/components/graph/network-modification-tree-model.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import { getLayoutedElements } from './layout';
+
+export default class NetworkModificationTreeModel {
+    treeElements = [];
+
+    updateLayout() {
+        this.treeElements = getLayoutedElements(this.treeElements);
+    }
+
+    addChild(element, parentId) {
+        // Add node
+        this.treeElements.push({
+            id: element.id,
+            type: element.type, // input node
+            data: { label: element.name, description: element.description },
+        });
+        // Add edge between node and its parent
+        this.treeElements.push({
+            id: 'e' + parentId + '-' + element.id,
+            source: parentId,
+            target: element.id,
+            type: 'smoothstep',
+        });
+        if (element.children) {
+            element.children.forEach((child) => {
+                this.addChild(child, element.id);
+            });
+        }
+    }
+
+    removeNodes(deletedNodes) {
+        deletedNodes.forEach((nodeId) => {
+            const edges = this.treeElements.filter(
+                (element) =>
+                    element.source === nodeId || element.target === nodeId
+            );
+            const filteredTreeElements = this.treeElements.filter(
+                (element) =>
+                    element.id !== nodeId &&
+                    element.source !== nodeId &&
+                    element.target !== nodeId
+            );
+            const edgesOfSource = edges.filter(
+                (element) => element.source === nodeId
+            );
+            const edgesOfTarget = edges.filter(
+                (element) => element.target === nodeId
+            );
+            edgesOfTarget.forEach((edgeOfTarget) => {
+                edgesOfSource.forEach((edgeOfSource) => {
+                    filteredTreeElements.push({
+                        id:
+                            'e' +
+                            edgeOfTarget.source +
+                            '-' +
+                            edgeOfSource.target,
+                        source: edgeOfTarget.source,
+                        target: edgeOfSource.target,
+                        type: 'smoothstep',
+                    });
+                });
+            });
+            this.treeElements = filteredTreeElements;
+        });
+    }
+
+    updateNodes(updatedNodes) {
```

Comment:
it only update the node name

Reply:
Yes, this is a first minimal version. updateNodes function will be modified in further versions to include other update types.